ts/features/itwallet/presentation/proximity/machine/actors.ts

```
@@ -163,30 +170,36 @@ export const createProximityActorsImplementation = (
             proximityDetails,
             verifierRequest: parsedRequest
           });
-        } catch (e) {
-          sendBack({
-            type: "device-error",
-            error: getError(e)
-          });
+        } catch (error) {
+          // Give some time to show the loading message
+          // and avoid glitches in the UI.
+          setTimeout(() => {
+            sendBack({
+              type: "device-error",
+              error
+            });
+          }, 500);
         }
       };
 
-      ISO18013_5.addListener("onDeviceConnecting", handleDeviceConnecting);
-      ISO18013_5.addListener("onDeviceConnected", handleDeviceConnected);
-      ISO18013_5.addListener(
-        "onDocumentRequestReceived",
-        handleDocumentRequestReceived
-      );
-      ISO18013_5.addListener("onDeviceDisconnected", handleDeviceDisconnected);
-      ISO18013_5.addListener("onError", handleError);
+      const listeners = [
+        ISO18013_5.addListener("onDeviceConnecting", handleDeviceConnecting),
+        ISO18013_5.addListener("onDeviceConnected", handleDeviceConnected),
+        ISO18013_5.addListener(
+          "onDocumentRequestReceived",
+          handleDocumentRequestReceived
+        ),
+        ISO18013_5.addListener(
+          "onDeviceDisconnected",
+          handleDeviceDisconnected
+        ),
+        ISO18013_5.addListener("onError", handleError)
+      ];
 
       return () => {
-        // Cleanup function
-        ISO18013_5.removeListener("onDeviceConnected");
-        ISO18013_5.removeListener("onDeviceConnecting");
-        ISO18013_5.removeListener("onDeviceDisconnected");
-        ISO18013_5.removeListener("onDocumentRequestReceived");
-        ISO18013_5.removeListener("onError");
+        // Remove event listeners
+        listeners.forEach(listener => listener.remove());
+        // Close the Bluetooth connection and clear all resources
         void ISO18013_5.close();
```

Should we ignore errors which might occur here?